target/linux/ath79/dts/qca9557_buffalo_bhr-4grv2.dts

```diff
@@ -103,15 +103,101 @@
 &mdio0 {
 	status = "okay";
 
-	phy0: ethernet-phy@0 {
+	switch0@10 {
+		compatible = "qca,qca8327";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
 		reg = <0>;
+		qca,sgmii-rxclk-falling-edge;
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			port@0 {
+				reg = <0>;
+				label = "cpu";
+				ethernet = <&eth1>;
+				phy-mode = "sgmii";
+				qca,sgmii-disable-pll;
+
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				label = "lan1";
+				phy-mode = "internal";
+				phy-handle = <&phy_port1>;
+			};
+
+			port@2 {
+				reg = <2>;
+				label = "lan2";
+				phy-mode = "internal";
+				phy-handle = <&phy_port2>;
+			};
+
+			port@3 {
+				reg = <3>;
+				label = "lan3";
+				phy-mode = "internal";
+				phy-handle = <&phy_port3>;
+			};
+
+			port@4 {
+				reg = <4>;
+				label = "lan4";
+				phy-mode = "internal";
+				phy-handle = <&phy_port4>;
+			};
+
+			port@5 {
+				reg = <5>;
+				label = "wan";
+				phy-mode = "internal";
+				phy-handle = <&phy_port5>;
+			};
+
+			port@6 {
+				reg = <6>;
+				label = "cpu";
+				ethernet = <&eth0>;
+				phy-mode = "rgmii-id";
+
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+		};
 
-		qca,ar8327-initvals = <
-		    0x04 0x80080080 /* PORT0 PAD MODE CTRL */
-		    0x0c 0x07600000 /* PORT6 PAD MODE CTRL */
-		    0x7c 0x0000007e /* PORT0_STATUS */
-		    0x94 0x0000007e /* PORT6 STATUS */
-		>;
+		mdio {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			phy_port1: phy@0 {
```

Comment:
I wonder whether these labels should correspond to the reg number?

Can there be a situation where there are multiple mdios and the names would break then (i.e. multiple phy_port0 etc.)?

Reply:
the mdio here is internal of the switch. I think we would encounter this situation only when the target have multiple switch. I think in that case we can consider a different naming?